lib/literature/uploaders/processor.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Literature.Uploaders.Processor do
+  @moduledoc false
+
+  alias Waffle.Transformations.Convert
+
+  def process(definition, version, {file, scope}) do
+    transform = definition.transform(version, {file, scope})
+    apply_transformation(file, transform)
+  end
+
+  defp apply_transformation(_, :skip), do: [{:ok, nil}]
+  defp apply_transformation(file, :noaction), do: [{:ok, file}]
+
+  defp apply_transformation(file, {cmd, conversion}) do
+    file
+    |> generates_image_sizes(conversion)
+    |> Enum.map(&Convert.apply(cmd, file, &1))
+  end
+
+  defp apply_transformation(file, {cmd, conversion, extension}) do
+    file
+    |> generates_image_sizes(conversion)
+    |> Enum.map(&Convert.apply(cmd, file, &1, extension))
+  end
+
+  # To generate image sizes from 100px until the image width
+  defp generates_image_sizes(%{path: path}, conversion) do
+    %{width: width} =
+      path
+      |> Mogrify.open()
+      |> Mogrify.verbose()
+
+    Range.new(100, width, 100)
```

Comment:
@christianlj i'm not sure i understand how this works i'm not sure what the conversion parameter is setting. How does it get stored in our bucket ? 

```
["-resize 100x100^ thumb", "-resize 200x200^ thumb", "-resize 300x300^ thumb",
 "-resize 400x400^ thumb", "-resize 500x500^ thumb", "-resize 600x600^ thumb",
 "-resize 700x700^ thumb", "-resize 800x800^ thumb", "-resize 900x900^ thumb"]
```